.github/workflows/build_rocks.yaml

```diff
@@ -83,33 +111,39 @@ jobs:
       contents: read
       packages: write
     needs: [get-rocks]
-    if: ${{ needs.get-rocks.outputs.rock-paths != '[]' }}
+    if: ${{ needs.get-rocks.outputs.rock-metas != '[]' }}
     strategy:
       matrix:
-        path: ${{ fromJSON(needs.get-rocks.outputs.rock-paths) }}
+        rock: ${{ fromJSON(needs.get-rocks.outputs.rock-metas) }}
```

Comment:
this is now a list of `metas` -- `[<meta>,<meta>,<meta>]`
a meta is an object that looks like:
```json
{
    name: <rock.name>,
    version: <rock.version>,
    path: <path to rockcraft.yaml>,
    arch:  <rock platform>,
    image: <expected image name>
}
```